File: binding/include/libqpen.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpen.html
pub const qpen = struct {

    /// New constructs a new QPen object.
    pub fn New() QtC.QPen {
        return qtc.QPen_new();
    }


    /// New2 constructs a new QPen object.
    pub fn New2(param1: i32) QtC.QPen {
        return qtc.QPen_new2(@intCast(param1));
    }


    /// New3 constructs a new QPen object.
    pub fn New3(color: ?*anyopaque) QtC.QPen {
        return qtc.QPen_new3(@ptrCast(color));
    }


    /// New4 constructs a new QPen object.
    pub fn New4(brush: ?*anyopaque, width: f64) QtC.QPen {
        return qtc.QPen_new4(@ptrCast(brush), width);
    }


    /// New5 constructs a new QPen object.
    pub fn New5(pen: ?*anyopaque) QtC.QPen {
        return qtc.QPen_new5(@ptrCast(pen));
    }


    /// New6 constructs a new QPen object.
    pub fn New6(brush: ?*anyopaque, width: f64, s: i32) QtC.QPen {
        return qtc.QPen_new6(@ptrCast(brush), width, @intCast(s));
    }


    /// New7 constructs a new QPen object.
    pub fn New7(brush: ?*anyopaque, width: f64, s: i32, c: i32) QtC.QPen {
        return qtc.QPen_new7(@ptrCast(brush), width, @intCast(s), @intCast(c));
    }


    /// New8 constructs a new QPen object.
    pub fn New8(brush: ?*anyopaque, width: f64, s: i32, c: i32, j: i32) QtC.QPen {
        return qtc.QPen_new8(@ptrCast(brush), width, @intCast(s), @intCast(c), @intCast(j));
    }


    pub fn OperatorAssign(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QPen_OperatorAssign(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPen_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn Style(self: ?*anyopaque, ) i32 {
        return qtc.QPen_Style(@ptrCast(self));
    }

    pub fn SetStyle(self: ?*anyopaque, style: i32) void {
        qtc.QPen_SetStyle(@ptrCast(self), @intCast(style));
    }

    pub fn DashPattern(self: ?*anyopaque, ) []const u8 {
        return qtc.QPen_DashPattern(@ptrCast(self));
    }

    pub fn SetDashPattern(self: ?*anyopaque, pattern: []const u8) void {
        qtc.QPen_SetDashPattern(@ptrCast(self), @ptrCast(pattern));
    }

    pub fn DashOffset(self: ?*anyopaque, ) f64 {
        return qtc.QPen_DashOffset(@ptrCast(self));
    }

    pub fn SetDashOffset(self: ?*anyopaque, doffset: f64) void {
        qtc.QPen_SetDashOffset(@ptrCast(self), doffset);
    }

    pub fn MiterLimit(self: ?*anyopaque, ) f64 {
        return qtc.QPen_MiterLimit(@ptrCast(self));
    }

    pub fn SetMiterLimit(self: ?*anyopaque, limit: f64) void {
        qtc.QPen_SetMiterLimit(@ptrCast(self), limit);
    }

    pub fn WidthF(self: ?*anyopaque, ) f64 {
        return qtc.QPen_WidthF(@ptrCast(self));
    }

    pub fn SetWidthF(self: ?*anyopaque, width: f64) void {
        qtc.QPen_SetWidthF(@ptrCast(self), width);
    }

    pub fn Width(self: ?*anyopaque, ) i32 {
        return qtc.QPen_Width(@ptrCast(self));
    }

    pub fn SetWidth(self: ?*anyopaque, width: i32) void {
        qtc.QPen_SetWidth(@ptrCast(self), width);
    }

    pub fn Color(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QPen_Color(@ptrCast(self));
    }

    pub fn SetColor(self: ?*anyopaque, color: ?*anyopaque) void {
        qtc.QPen_SetColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QPen_Brush(@ptrCast(self));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QPen_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn IsSolid(self: ?*anyopaque, ) bool {
        return qtc.QPen_IsSolid(@ptrCast(self));
    }

    pub fn CapStyle(self: ?*anyopaque, ) i32 {
        return qtc.QPen_CapStyle(@ptrCast(self));
    }

    pub fn SetCapStyle(self: ?*anyopaque, pcs: i32) void {
        qtc.QPen_SetCapStyle(@ptrCast(self), @intCast(pcs));
    }

    pub fn JoinStyle(self: ?*anyopaque, ) i32 {
        return qtc.QPen_JoinStyle(@ptrCast(self));
    }

    pub fn SetJoinStyle(self: ?*anyopaque, pcs: i32) void {
        qtc.QPen_SetJoinStyle(@ptrCast(self), @intCast(pcs));
    }

    pub fn IsCosmetic(self: ?*anyopaque, ) bool {
        return qtc.QPen_IsCosmetic(@ptrCast(self));
    }

    pub fn SetCosmetic(self: ?*anyopaque, cosmetic: bool) void {
        qtc.QPen_SetCosmetic(@ptrCast(self), cosmetic);
    }

    pub fn OperatorEqual(self: ?*anyopaque, p: ?*anyopaque) bool {
        return qtc.QPen_OperatorEqual(@ptrCast(self), @ptrCast(p));
    }

    pub fn OperatorNotEqual(self: ?*anyopaque, p: ?*anyopaque) bool {
        return qtc.QPen_OperatorNotEqual(@ptrCast(self), @ptrCast(p));
    }

    pub fn OperatorQVariant(self: ?*anyopaque, ) QtC.QVariant {
        return qtc.QPen_OperatorQVariant(@ptrCast(self));
    }

    pub fn IsDetached(self: ?*anyopaque, ) bool {
        return qtc.QPen_IsDetached(@ptrCast(self));
    }

    pub fn DataPtr(self: ?*anyopaque, ) QExplicitlySharedDataPointer<QPenPrivate> {
        return @ptrCast(qtc.QPen_DataPtr(@ptrCast(self)));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPen_Delete(@ptrCast(self));
    }
};

File: binding/src/libqpen.cpp
#include <QBrush>
#include <QColor>
#include <QPen>
#include <QVariant>
#include <qpen.h>
#include "libqpen.h"
#include "libqpen.hxx"

QPen* QPen_new() {
	 return new QPen();
}

QPen* QPen_new2(int param1) {
	 return new QPen(static_cast<Qt::PenStyle>(param1));
}

QPen* QPen_new3(const QColor* color) {
	 return new QPen(*color);
}

QPen* QPen_new4(const QBrush* brush, double width) {
	 return new QPen(*brush, width);
}

QPen* QPen_new5(const QPen* pen) {
	 return new QPen(*pen);
}

QPen* QPen_new6(const QBrush* brush, double width, int s) {
	 return new QPen(*brush, width, static_cast<Qt::PenStyle>(s));
}

QPen* QPen_new7(const QBrush* brush, double width, int s, int c) {
	 return new QPen(*brush, width, static_cast<Qt::PenStyle>(s), static_cast<Qt::PenCapStyle>(c));
}

QPen* QPen_new8(const QBrush* brush, double width, int s, int c, int j) {
	 return new QPen(*brush, width, static_cast<Qt::PenStyle>(s), static_cast<Qt::PenCapStyle>(c), static_cast<Qt::PenJoinStyle>(j));
}

void QPen_OperatorAssign(QPen* self, const QPen* pen) {
	self->operator=(*pen);
}

void QPen_Swap(QPen* self, QPen* other) {
	self->swap(*other);
}

int QPen_Style(const QPen* self) {
	return self->style();
}

void QPen_SetStyle(QPen* self, int style) {
	self->setStyle(static_cast<Qt::PenStyle>(style));
}

libqt_list QPen_DashPattern(const QPen* self) {
	return self->dashPattern();
}

void QPen_SetDashPattern(QPen* self, const libqt_list pattern) {
	self->setDashPattern(*pattern);
}

double QPen_DashOffset(const QPen* self) {
	return self->dashOffset();
}

void QPen_SetDashOffset(QPen* self, double doffset) {
	self->setDashOffset(doffset);
}

double QPen_MiterLimit(const QPen* self) {
	return self->miterLimit();
}

void QPen_SetMiterLimit(QPen* self, double limit) {
	self->setMiterLimit(limit);
}

double QPen_WidthF(const QPen* self) {
	return self->widthF();
}

void QPen_SetWidthF(QPen* self, double width) {
	self->setWidthF(width);
}

int QPen_Width(const QPen* self) {
	return self->width();
}

void QPen_SetWidth(QPen* self, int width) {
	self->setWidth(width);
}

QColor* QPen_Color(const QPen* self) {
	return new QColor(self->color());
}

void QPen_SetColor(QPen* self, const QColor* color) {
	self->setColor(*color);
}

QBrush* QPen_Brush(const QPen* self) {
	return new QBrush(self->brush());
}

void QPen_SetBrush(QPen* self, const QBrush* brush) {
	self->setBrush(*brush);
}

bool QPen_IsSolid(const QPen* self) {
	return self->isSolid();
}

int QPen_CapStyle(const QPen* self) {
	return self->capStyle();
}

void QPen_SetCapStyle(QPen* self, int pcs) {
	self->setCapStyle(static_cast<Qt::PenCapStyle>(pcs));
}

int QPen_JoinStyle(const QPen* self) {
	return self->joinStyle();
}

void QPen_SetJoinStyle(QPen* self, int pcs) {
	self->setJoinStyle(static_cast<Qt::PenJoinStyle>(pcs));
}

bool QPen_IsCosmetic(const QPen* self) {
	return self->isCosmetic();
}

void QPen_SetCosmetic(QPen* self, bool cosmetic) {
	self->setCosmetic(cosmetic);
}

bool QPen_OperatorEqual(const QPen* self, const QPen* p) {
	return self->operator==(*p);
}

bool QPen_OperatorNotEqual(const QPen* self, const QPen* p) {
	return self->operator!=(*p);
}

QVariant* QPen_OperatorQVariant(const QPen* self) {
	return new QVariant(self->operator QVariant());
}

bool QPen_IsDetached(QPen* self) {
	return self->isDetached();
}

QExplicitlySharedDataPointer<QPenPrivate>* QPen_DataPtr(QPen* self) {
	return self->data_ptr();
}

void QPen_Delete(QPen* self) {
    delete self;
}

